.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,32 @@
+name: On Pull Request
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    runs-on: self-hosted
+
+    # The cimg-mvn-cache is an image containing a .m2 folder warmed-up
+    # with common Jahia dependencies. Using this prevents maven from
+    # downloading the entire world when building.
+    # More on https://github.com/Jahia/cimg-mvn-cache
+    container:
+      image: jahia/cimg-mvn-cache:ga_cimg_openjdk_11.0.20-node
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '11.0.19'
+
+      - name: Build with Gradle
+        run: ./gradlew shadowJar
```

Comment:
Should there be something like saving the artifact after the build ?

Reply:
You're referring to the https://github.com/actions/upload-artifact action, right?
I don't really know the rules we have in place regarding the upload of the generated artifacts, but this step can be added here, no problem, I'll make the change

Reply:
I added it below, let me know if that's what you had in mind, thanks!